src/main/java/org/terasology/web/io/JsonSession.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2017 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.web.io;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonSyntaxException;
+import org.terasology.config.Config;
+import org.terasology.entitySystem.entity.EntityManager;
+import org.terasology.identity.storageServiceClient.BigIntegerBase64Serializer;
+import org.terasology.web.EngineRunner;
+import org.terasology.web.authentication.AuthenticationFailedException;
+import org.terasology.web.authentication.AuthenticationHandshakeHandler;
+import org.terasology.web.authentication.AuthenticationHandshakeHandlerImpl;
+import org.terasology.web.authentication.ClientAuthenticationMessage;
+import org.terasology.web.authentication.HandshakeHello;
+import org.terasology.web.client.HeadlessClient;
+import org.terasology.web.client.HeadlessClientFactory;
+
+import java.math.BigInteger;
+
+public class JsonSession {
+
+    private static final Gson GSON = new GsonBuilder()
+            .registerTypeHierarchyAdapter(BigInteger.class, BigIntegerBase64Serializer.getInstance())
```

Comment:
I agree that the Hierarchy is useless here since there are no subclasses and I can't remember why I used it, most likely misclicked an IDE autocompletion suggestion.